bundles/org.dataflowanalysis.analysis.converter/src/org/dataflowanalysis/analysis/converter/ProcessJSON.java

```diff
@@ -0,0 +1,366 @@
+package org.dataflowanalysis.analysis.converter;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.dataflowanalysis.analysis.converter.microsecend.*;
+import org.dataflowanalysis.analysis.converter.webdfd.*;
+import org.dataflowanalysis.dfd.datadictionary.*;
+import org.dataflowanalysis.dfd.dataflowdiagram.*;
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.emf.ecore.resource.ResourceSet;
+import org.eclipse.emf.ecore.resource.impl.ResourceImpl;
+import org.eclipse.emf.ecore.resource.impl.ResourceSetImpl;
+import org.eclipse.emf.ecore.xmi.XMLResource;
+import org.eclipse.emf.ecore.xmi.impl.XMLResourceFactoryImpl;
+
+public class ProcessJSON {
+	private dataflowdiagramFactory dfdFactory;
+	private datadictionaryFactory ddFactory;
+	private ResourceSet rs;
+	
+	private Map<String, Node> nodesMap;
+	private Map<Node,List<String>> nodeToLabelNames;
+	private Map<String,LabelType> labelTypeMap;
+	private Map<String, Label> labelMap;
+			
+	public ProcessJSON() {
+		dfdFactory = dataflowdiagramFactory.eINSTANCE;
+		ddFactory = datadictionaryFactory.eINSTANCE;
+		rs = new ResourceSetImpl();
+		
+		nodesMap = new HashMap<>();
+		nodeToLabelNames = new HashMap<>();
+		labelTypeMap = new HashMap<>();
+		labelMap = new HashMap<>();
+	}
+	
+	@SuppressWarnings({ "rawtypes", "unchecked" })
+	public Resource createAndAddResource(String outputFile, String[] fileextensions, ResourceSet rs) {
+	     for (String fileext : fileextensions) {
+	        rs.getResourceFactoryRegistry().getExtensionToFactoryMap().put(fileext, new XMLResourceFactoryImpl());
+	     }		
+	     URI uri = URI.createFileURI(outputFile);
+	     Resource resource = rs.createResource(uri);
+	     ((ResourceImpl)resource).setIntrinsicIDToEObjectMap(new HashMap());
+	     return resource;
+	  }
+	
+	@SuppressWarnings({"unchecked", "rawtypes"})
+	public void saveResource(Resource resource) {
+	     Map saveOptions = ((XMLResource)resource).getDefaultSaveOptions();
+	     saveOptions.put(XMLResource.OPTION_CONFIGURATION_CACHE, Boolean.TRUE);
+	     saveOptions.put(XMLResource.OPTION_USE_CACHED_LOOKUP_TABLE, new ArrayList());
+	     try {
+	        resource.save(saveOptions);
+	     } 
+	     catch (IOException e) {
+	        throw new RuntimeException(e);
+	     }
+	}
```

Comment:
Why `@SuppressWarnings` here? Is it due to the casting of `XMLResource`? It should be possible to do without unsafe casting between the types